### ISSUE TITLE
Define and bundle monospace font explicitly

### ISSUE THREAD

**Issue by GergelyKalmar** [readthedocs/sphinx_rtd_theme#1435]
### Problem

The monospace font is currently specified as a list of system fonts, unlike the body text, where the fonts are bundled with the theme. It is possible to override this behavior via a custom Sphinx style file, but it is relatively complicated because the additional fonts also need to be installed and picked up during build time, also, a bunch of elements do not use the right font family by default (see an override here: https://github.com/logikal-io/logikal-docs/blob/main/logikal_docs/static/css/logikal_docs.css#L4-L35). Using something like Roboto Mono would go well with the current text fonts and is also relatively similar to the default system fonts that are currently picked up. Issue https://github.com/readthedocs/sphinx_rtd_theme/issues/1295 is also related to this point. Standard Roboto Mono files can be found [here](https://github.com/logikal-io/logikal-docs/tree/main/logikal_docs/static/css/fonts).

### Reproducible Project

The URL https://docs.logikal.io/pyorbs/2.0.0/ shows a page where the Roboto Mono fonts are used for monospace text, while https://docs.logikal.io/pyorbs/1.0.0/ is an example where system fonts are used. In my case the picked up system font (Liberation Mono) looks almost exactly the same as Roboto Mono (apart from some smaller differences in terminals and descenders).

**Comment by benjaoming** [comment]
Thanks for the report! How does the broken behavior look in a screenshot?

**Comment by GergelyKalmar** [comment]
Well, that's the point, the monospace font looks different for each user right now depending on their OS and user fonts. Here is how the page https://docs.logikal.io/pyorbs/1.0.0/ looks on my system:

![screenshot](https://user-images.githubusercontent.com/5601876/218465438-ffa33f5b-3189-4a36-bfcd-e9758de63dfd.png)

Depending on your system fonts the code snippets might look slightly or totally different for you.

**Comment by benjaoming** [comment]
That's fair, I just wanted to know how bad it gets, like if there was a known case where it's really off :+1: It's nice to get fixed anyhow.

I was on another Sphinx project with a theme with exactly the same setup, [the RTD blog](https://blog.readthedocs.com/): Default text font was embedded but monospace wasn't and the embedded font had quite a small size when rendered which gave a relatively big mismatch with the system monospace.

**Comment by GergelyKalmar** [comment]
Yes, that's a good example too, I see the same on the RTD blog. I guess it's not a very difficult fix, although if you also want to make the font families easier to configure (perhaps by using CSS variables here as well), that might be a bit bigger work.

**Comment by ia0** [comment]
> I just wanted to know how bad it gets, like if there was a known case where it's really off

I believe it's really off on Android. Consider https://rocq-prover.org/doc/V9.0.0/refman/addendum/implicit-coercions.html#id1

Here is how it looks like on Chrome on Android (notice how the subscripts are identically rendered as blocks):

<img width="540" height="963" alt="Image" src="https://github.com/user-attachments/assets/3c6ed0eb-b836-418f-8e9e-ccf4eb5a940f" />

Here is how it looks like on Chrome on Linux (notice how the subscripts are correctly rendered):

<img width="1096" height="737" alt="Image" src="https://github.com/user-attachments/assets/f7f3b9d1-8b99-4119-89f6-262bd3365369" />
